src/lib/storage/column_encoders/base_column_encoder.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <boost/hana/type.hpp>
+
+#include <memory>
+
+#include "storage/base_value_column.hpp"
+#include "storage/encoded_columns/column_encoding_type.hpp"
+
+#include "all_type_variant.hpp"
+#include "resolve_type.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+namespace hana = boost::hana;
+
+/**
+ * @brief Base class of all column encoders
```

Comment:
Do you have an extensive comment on how everything works somewhere? This could use a pointer to that comment (or a wiki entry)

Reply:
I’ll add one.

Reply:
Edit: I think we have to talk about this together. I have worked too long on this code to know what should be explained and what not.